doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -552,90 +517,78 @@ the conversion in hypotheses :n:`{+ @ident}`.
 
-.. tacv:: simpl @pattern
+.. tacn:: cbn {? @reductions } {? @occurrences }
 
-   This applies :tacn:`simpl` only to the subterms matching
-   :n:`@pattern` in the current goal.
+   :tacn:`cbn` tactic was intended to be a more principled, faster and more
+   predictable replacement for :tacn:`simpl`.
 
-.. tacv:: simpl @pattern at {+ @natural}
+   :tacn:`cbn` tactic accepts the same reductions as :tacn:`cbv` and
+   :tacn:`lazy`. The behavior can be tuned using the :cmd:`Arguments` command.
 
-   This applies :tacn:`simpl` only to the :n:`{+ @natural}` occurrences of the subterms
-   matching :n:`@pattern` in the current goal.
+   The main difference between :tacn:`cbn` and :tacn:`simpl` is that
+   :tacn:`cbn` reduces whether it will be able to reuse the constant or not:
+   :g:`succ t` is reduced to :g:`S t`.
 
    .. exn:: Too few occurrences.
       :undocumented:
 
-.. tacv:: simpl @qualid
-          simpl @string
-
-   This applies :tacn:`simpl` only to the applicative subterms whose head occurrence
-   is the unfoldable constant :n:`@qualid` (the constant can be referred to by
-   its notation using :n:`@string` if such a notation exists).
-
-.. tacv:: simpl @qualid at {+ @natural}
-          simpl @string at {+ @natural}
-
-   This applies :tacn:`simpl` only to the :n:`{+ @natural}` applicative subterms whose
-   head occurrence is :n:`@qualid` (or :n:`@string`).
+   .. flag:: Debug RAKAM
 
-.. flag:: Debug RAKAM
+      This flag makes :tacn:`cbn` print various debugging information.
+      ``RAKAM`` is the Refolding Algebraic Krivine Abstract Machine.
 
-   This flag makes :tacn:`cbn` print various debugging information.
-   ``RAKAM`` is the Refolding Algebraic Krivine Abstract Machine.
+.. tacn:: unfold {+, @reference_occs } {? @occurrences }
 
-.. tacn:: unfold @qualid
-   :name: unfold
-
-   This tactic applies to any goal. The argument qualid must denote a
-   defined transparent constant or local definition (see
-   :ref:`gallina-definitions` and
-   :ref:`vernac-controlling-the-reduction-strategies`). The tactic
-   :tacn:`unfold` applies the :math:`\delta` rule to each occurrence
-   of the constant to which :n:`@qualid` refers in the current goal
-   and then replaces it with its :math:`\beta\iota\zeta`-normal form.
-   Use the general reduction tactics if you want to avoid this final
-   reduction, for instance :n:`cbv delta [@qualid]`.
+   Applies the :math:`\delta` rule to each occurrence
+   of the constants specified by each :n:`@reference_occs`
+   and replaces them with their :math:`\beta\iota\zeta`-normal form.
+   Each :n:`@reference_occs` must be a defined transparent constant or local
+   definition (see :ref:`gallina-definitions` and
+   :ref:`controlling-the-reduction-strategies`).
+   Use the general reduction tactics if you want only the to apply the
+   :math:`\delta` rule, for example :n:`cbv delta [@reference]`.
 
    .. exn:: Cannot coerce @qualid to an evaluable reference.
```

Comment:
The output of the example shows that this error message no longer exists. The two examples raise "Cannot turn inductive ... into an evaluable reference." and "... is opaque". Given that the error messages are clearer now, the examples are not needed as much. That being said, it might be a good idea to generally strive to provide one example per documented error message.

Reply:
The message is still in the code in `Tacred.error_not_evaluable`.  Don't know if it can still be generated.

I'm not so enthusiastic about having an example for each error message: having more working examples for tactics/commands seems more important.

Reply:
> I'm not so enthusiastic about having an example for each error message: having more working examples for tactics/commands seems more important.

I wasn't saying that we should work on this but it was more as a reasoning to conclude that even if the examples are not that important anymore, it's worth to keep them.

Reply:
Yeah, I thought that might be what you were getting at.

Reply:
Examples of commands that generate this error:
```coq
Hint Transparent unit.
Typeclasses Transparent unit.
```

I'm not sure if the error can be hit in Ltac.

Reply:
This issue is still there: this tactic does not raise this error anymore but two new error messages instead. The documentation should be updated to account for this (each example corresponds to one of the new error message).